### ISSUE TITLE
When deleting a subclassed entity, all relations declared in the base class are not severed

### ISSUE THREAD

**Issue by alexis-y** [Breeze/breeze.js#168]
According to the [docs](http://breeze.github.io/doc-js/api-docs/classes/EntityAspect.html#method_setDeleted): 

> This both marks the entity as being scheduled for deletion during the next 'Save' call but also removes the entity from all of its related entities.

However, it doesn't do so for relations declared in the base class of an entity. This happens no matter the multiplicity of both ends of the relation.
### Scenario

Assume the following entities:
- ChildBase (abstract, with NP "Parent", scalar)
- ChildConcrete (subclass of ChildBase)
- Parent (with NP "Children", non-scalar)
### Reproduce

Now, retrieve a parent with it's "Children" property loaded, and 1 or more children (of type ChildConcrete). 

```
breeze.EntityQuery.from("Parents").expand("Children").take(1).execute(db).then(function(response) {
  var parent = response.results[0];
  var child = parent.children[0];

  child.entityAspect.setDeleted(); // Moment of truth

  console.log(parent.children.includes(child));  // Is the deleted child still in the parent? True
});
```
### Probable cause

I've tracked the issue down to EntityAspect, removeFromRelationsCore(entity):

```
function removeFromRelationsCore(entity) {
  entity.entityType.navigationProperties.forEach(function (np) {
    // yada yada
  });
}
```

entityType.navigationProperties only has the NPs declared in this type, _and not in it's base class_.
### Workaround

Manually sever the association after deleting the child.

**Comment by steveschmitt** [comment]
I _think_ this was fixed in version 1.5.11 and later.  Please get the latest version (1.5.16 as of today) and try again.
Thanks!

**Comment by alexis-y** [comment]
Sorry everyone, I further tracked the issue to the way I was creating the properties. I create some additional NPs after declaring the concrete entities, and did not realize that the properties of the base type are actually cloned in the child types. When used normally, **this issue doesn't happen**.

I still find it _weird_ that inherited properties are actually contained in the subtype, as it leads to this kind of behavior where if changes are made after a subtype is declared, the changes are not propagated.
